crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -1509,13 +1512,30 @@ mod tests {
         Vec<SealedBlock>,
         Vec<Vec<Receipt>>,
     )> {
-        let (database_blocks, in_memory_blocks) = random_blocks(
+        let (mut database_blocks, mut in_memory_blocks) = random_blocks(
             rng,
             database_blocks,
             in_memory_blocks,
             block_range_params.requests_count,
             block_range_params.withdrawals_count,
         );
+
+        for block in &mut database_blocks {
+            *block = SealedBlock {
+                ommers: vec![Header::default(); 7],
+                body: (0..block_range_params.tx_count.end).map(|_| random_signed_tx(rng)).collect(),
+                ..block.clone()
+            };
+        }
+
+        for block in &mut in_memory_blocks {
+            *block = SealedBlock {
+                ommers: vec![Header::default(); 7],
+                body: (0..block_range_params.tx_count.end).map(|_| random_signed_tx(rng)).collect(),
+                ..block.clone()
+            };
+        }
```

Comment:
`random_blocks` should generate random transactions on its own if we pass `block_range_params` directly into it as an argument. `random_blocks_range` will take care of this.

Reply:
Sorry for the back and forth... I think this should be good, tell me what you think.

But actually I feel like this test setup is a bit messy, given the large amount of tests that depend on it, maybe we should make an issue to implement a dedicated test utils structure on a file next to it in order to clean up this part and handle all the scenarios properly, what do you think?

Reply:
yeah, agreed there are a lot of tests, but you can reduce the amount 3x by collapsing (database, in-memory, not found) tests into one, and then you'll have just one setup for all of them.